Matey/Controllers/HomeViewController.swift

```diff
@@ -8,90 +8,105 @@
-  @IBOutlet weak var labelWelcome: UILabel!
-  @IBOutlet weak var labelTotalBalance: UILabel!
-  @IBOutlet weak var labelTotalBalanceAmount: UILabel!
-  @IBOutlet weak var labelLend: UILabel!
-  @IBOutlet weak var labelLendAmount: UILabel!
-  @IBOutlet weak var labelBorrow: UILabel!
-  @IBOutlet weak var labelBorrowAmount: UILabel!
-  @IBOutlet weak var labelHeader: UILabel!
-  var plusButton: UIButton!
-
-  // MARK: - Properties
-
-  let coreDataManager = CoreDataManager()
-
-  // MARK: - Life Cycle
-
-  override func viewDidLoad() {
-    super.viewDidLoad()
-    
-    uiInit()
-    setupPlusButton()
-  }
-
-  // MARK: - Functions
-
-  private func uiInit() {
-    title = "Home"
-    labelName.text = "Hi Fırat"
-    labelName.font = UIFont.boldSystemFont(ofSize: 32)
-    labelWelcome.text = "welcome back"
-    labelWelcome.font = UIFont.systemFont(ofSize: 24)
-    labelWelcome.textColor = .secondaryLabel
-    labelTotalBalance.text = "Total Balance"
-    labelTotalBalance.font = UIFont.systemFont(ofSize: 16)
-    labelTotalBalance.textColor = .secondaryLabel
-    labelTotalBalanceAmount.text = "8,400 ₺"
-    labelTotalBalanceAmount.font = UIFont.boldSystemFont(ofSize: 24)
-    labelLend.text = "Lend"
-    labelLend.font = UIFont.systemFont(ofSize: 16)
-    labelLend.textColor = .secondaryLabel
-    labelLendAmount.text = "10,000 ₺"
-    labelLendAmount.font = UIFont.boldSystemFont(ofSize: 24)
-    labelLendAmount.textColor = UIColor(displayP3Red: 47/255, green: 140/255, blue: 79/255, alpha: 1.0)
-    labelBorrow.text = "Borrow"
-    labelBorrow.font = UIFont.systemFont(ofSize: 16)
-    labelBorrow.textColor = .secondaryLabel
-    labelBorrowAmount.text = "1,600 ₺"
-    labelBorrowAmount.font = UIFont.boldSystemFont(ofSize: 24)
-    labelBorrowAmount.textColor = .red
-    labelHeader.text = "Where is my Money?"
-    labelHeader.font = UIFont.boldSystemFont(ofSize: 24)
-    labelHeader.textColor = UIColor(displayP3Red: 47/255, green: 140/255, blue: 79/255, alpha: 1.0)
-  }
-
-  func setupPlusButton() {
-          plusButton = UIButton(frame: CGRect(x: 0, y: 0, width: 64, height: 64))
-          var menuButtonFrame = plusButton.frame
+    // MARK: - UI Elements
+    @IBOutlet weak private var labelName: UILabel!
+    @IBOutlet weak private var labelWelcome: UILabel!
+    @IBOutlet weak private var labelTotalBalance: UILabel!
+    @IBOutlet weak private var labelTotalBalanceAmount: UILabel!
+    @IBOutlet weak private var labelLend: UILabel!
+    @IBOutlet weak private var labelLendAmount: UILabel!
+    @IBOutlet weak private var labelBorrow: UILabel!
+    @IBOutlet weak private var labelBorrowAmount: UILabel!
+    @IBOutlet weak private var labelHeader: UILabel!
+    private var plusButton: UIButton!
+
+    // MARK: - Properties
+
+    var viewModel: HomeViewModelInterface?
+
+    // MARK: - Life Cycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        viewModel = HomeViewModel(view: HomeViewController())
```

Comment:
```suggestion
        viewModel = HomeViewModel(view: self)
```
Sürekli bir create döngüsü oluşturulduğundan burası sıkıntılı gibi geldi. Safa inceledik böyle bir çözüm bulduk umarım işe yarar.